File: Intercom/app/src/main/java/com/twiceyuan/intercom/common/DP.java
package com.twiceyuan.intercom.common;

import android.content.Context;
import android.util.DisplayMetrics;

import com.twiceyuan.intercom.App;

public class DP {

    public static int dp2px(float dpValue) {
        final float scale = App.get().getResources().getDisplayMetrics().density;
        return (int) (dpValue * scale + 0.5f);
    }

    public static int px2dp(float pxValue) {
        final float scale = App.get().getResources().getDisplayMetrics().density;
        return (int) (pxValue / scale + 0.5f);
    }

    /**
     * 获取屏幕宽度
     */
    public static int screenWidth(Context context) {
        DisplayMetrics dm = context.getResources().getDisplayMetrics();
        return dm.widthPixels;
    }

    /**
     * 获取屏幕高度
     */
    public static int screenHeight(Context context) {
        DisplayMetrics dm = context.getResources().getDisplayMetrics();
        return dm.heightPixels;
    }
}

File: Intercom/app/src/main/java/com/twiceyuan/intercom/common/DP.kt
package com.twiceyuan.intercom.common

import android.content.Context
import android.util.DisplayMetrics

import com.twiceyuan.intercom.App

object DP {

    fun dp2px(dpValue: Float): Int {
        val scale = App.get().resources.displayMetrics.density
        return (dpValue * scale + 0.5f).toInt()
    }

    fun px2dp(pxValue: Float): Int {
        val scale = App.get().resources.displayMetrics.density
        return (pxValue / scale + 0.5f).toInt()
    }

    /**
     * 获取屏幕宽度
     */
    fun screenWidth(context: Context): Int {
        val dm = context.resources.displayMetrics
        return dm.widthPixels
    }

    /**
     * 获取屏幕高度
     */
    fun screenHeight(context: Context): Int {
        val dm = context.resources.displayMetrics
        return dm.heightPixels
    }
}
